legato-api/src/main/java/org/opendaylight/legato/util/LegatoUtils.java

```diff
@@ -0,0 +1,386 @@
+						.read(store, l2cpEec_ProfileId);
+				return l2cpEecProfileFuture.checkedGet();
+
+			case LegatoConstants.L2CP_PEERING_PROFILES:
+				final InstanceIdentifier<org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.l2cp.peering.profiles.Profile> l2cpPeering_ProfileId = child
+						.firstIdentifierOf(
+								org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.l2cp.peering.profiles.Profile.class);
+				final CheckedFuture<Optional<org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.l2cp.peering.profiles.Profile>, ReadFailedException> l2cpPeeringProfileFuture = read
+						.read(store, l2cpPeering_ProfileId);
+				return l2cpPeeringProfileFuture.checkedGet();
+
+			case LegatoConstants.EEC_PROFILES:
+				final InstanceIdentifier<org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.eec.profiles.Profile> eecProfileId = child
+						.firstIdentifierOf(
+								org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.eec.profiles.Profile.class);
+				final CheckedFuture<Optional<org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.eec.profiles.Profile>, ReadFailedException> eecProfileFuture = read
+						.read(store, eecProfileId);
+				return eecProfileFuture.checkedGet();
+				
+			case LegatoConstants.CMP_PROFILES:
+				final InstanceIdentifier<org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.color.mapping.profiles.Profile> cmpProfileId = child
+						.firstIdentifierOf(
+								org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.color.mapping.profiles.Profile.class);
+				final CheckedFuture<Optional<org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.color.mapping.profiles.Profile>, ReadFailedException> cmpProfileFuture = read
+						.read(store, cmpProfileId);
+				return cmpProfileFuture.checkedGet();
+
+			default:
+				LOG.info("IN DEFAULT CASE :  NO MATCH");
+			}
+		} catch (final ReadFailedException e) {
+			LOG.error("Unable to read node ", e);
+		}
+		return Optional.absent();
+	}
+
+	@SuppressWarnings("deprecation")
+	public static void deleteFromOperationalDB(InstanceIdentifier<?> nodeIdentifier, DataBroker dataBroker) {
+		WriteTransaction tx = dataBroker.newWriteOnlyTransaction();
+		tx.delete(LogicalDatastoreType.OPERATIONAL, nodeIdentifier);
+
+		try {
+			tx.submit().checkedGet();
+		} catch (org.opendaylight.controller.md.sal.common.api.data.TransactionCommitFailedException e) {
+			LOG.error("Error in deleteFromOperationalDB(). Err: ", e);
+		}
+	}
+
+	public static <T extends DataObject> void addToOperationalDB(T typeOfProfile, InstanceIdentifier<T> profilesTx,
+			DataBroker dataBroker) {
+		WriteTransaction tx = dataBroker.newWriteOnlyTransaction();
+		tx.merge(LogicalDatastoreType.OPERATIONAL, profilesTx, typeOfProfile);
+		try {
+			tx.submit().checkedGet();
```

Comment:
as you are not interested in result you do not need to check for submit result. you can use ``Futures.addCallback(``